### PR TITLE
TDQ-13660 TDQ-13768 throw error messages when switch group before run…

### PR DESCRIPTION
**Pull request by xqliu-talend**
… the job

### CODE REVIEW

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/AnalysisExecutor.java [hunk 1]
```diff
@@ -594,23 +595,62 @@ protected void initializeIndicators(Analysis analysis) {
         // TDQ-8202~
     }
 
-    protected boolean changeCatalog(String catalogName, java.sql.Connection connection) {
+    /**
+     * return true if the connection need to set the catalog, otherwise return false.
+     * <p>
+     * The list of database connection which need not to change the catalog:
```
Comment (msjian) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/AnalysisExecutor.java]: The list of databases which don't need to change the catalog:

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/AnalysisExecutor.java [hunk 2]
```diff
@@ -594,23 +595,62 @@ protected void initializeIndicators(Analysis analysis) {
...
+     * 
+     * @return need to change the catalog or not
+     */
+    protected boolean needChangeCatalog(java.sql.Connection connection) {
+        boolean result = true;
         try {
             DatabaseMetaData metadata = ExtractMetaDataUtils.getInstance().getConnectionMetadata(connection);
-            if (!(ConnectionUtils.isOdbcMssql(connection) || ConnectionUtils.isOdbcOracle(connection)
+            result = !(ConnectionUtils.isOdbcMssql(connection) || ConnectionUtils.isOdbcOracle(connection)
                     || ConnectionUtils.isOdbcProgress(connection) || ConnectionUtils.isOdbcTeradata(connection)
```
Comment (msjian) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/AnalysisExecutor.java]: the above 4  methods are isOdbcXXX(), i don't think they are exactly right, what's about isMssql(), isPostgresql(), isTeradata().....?
and does there exist other database types?

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/TableAnalysisSqlExecutor.java [hunk 3]
```diff
@@ -429,8 +429,10 @@ public void run() {
     private boolean executeIndicator(Indicator indicator, Connection connection) {
         // set the connection's catalog
         String catalogName = getCatalogOrSchemaName(indicator.getAnalyzedElement());
-        if (catalogName != null) { // check whether null argument can be given
-            changeCatalog(catalogName, connection);
+        if (needChangeCatalog(connection)) {
+            if (catalogName != null) { // check whether null argument can be given
```
Comment (msjian) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/TableAnalysisSqlExecutor.java]: change to
if (catalogName != null && needChangeCatalog(connection)) {

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/AnalysisExecutor.java (hunk 2 above)
Comment (xqliu-talend) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/dq/analysis/AnalysisExecutor.java]: 1. for these 4 database types, don't need to change the catalog only using odbc to connect them
2. there should exist other database types which don't need to change the catalog, let add them into this list when we find them later

**Pull request merged**